sw/host/opentitanlib/src/test_utils/otp_ctrl.rs

```diff
@@ -0,0 +1,213 @@
+    const POLL_DELAY: Duration = Duration::from_millis(5);
+
+    /// Read a parameter from OTP into an output buffer.
+    pub fn read_param(jtag: &dyn Jtag, param: DamParam, out_buf: &mut [u32]) -> OtpParamResult<()> {
+        let OtpParamMmap { byte_addr, size } = param.mmap();
+
+        if mem::size_of_val(out_buf) < size as usize {
+            let err = OtpParamError::BufSize {
+                buf_size: mem::size_of_val(out_buf),
+                param_size: size,
+            };
+            return Err(err);
+        }
+
+        let Partition { access_granule, .. } = param.partition();
+
+        match access_granule {
+            Granularity::B32 => {
+                let out_iter = out_buf.iter_mut().take(size as usize);
+                for (idx, out_word) in out_iter.enumerate() {
+                    let addr = byte_addr + (idx * mem::size_of::<u32>()) as u32;
+                    let [lower, _] = Self::dmi_read(jtag, addr, access_granule)?;
+                    *out_word = lower;
+                }
+            }
+            Granularity::B64 => {
+                let out_iter = out_buf.chunks_mut(2).take(size as usize / 2);
+                for (idx, out_words) in out_iter.enumerate() {
+                    let addr = byte_addr + (idx * mem::size_of::<u64>()) as u32;
+                    let otp_words = Self::dmi_read(jtag, addr, access_granule)?;
+                    out_words[0] = otp_words[0];
+                    out_words[1] = otp_words[1];
+                }
+            }
+        }
+
+        Ok(())
+    }
+
+    /// Write a value from a buffer to an OTP parameter.
+    pub fn write_param(jtag: &dyn Jtag, param: DamParam, data: &[u32]) -> OtpParamResult<()> {
+        let OtpParamMmap { byte_addr, size } = param.mmap();
+
+        if mem::size_of_val(data) > size as usize {
+            let err = OtpParamError::BufSize {
+                buf_size: mem::size_of_val(data),
+                param_size: size,
+            };
+            return Err(err);
+        }
+
+        let Partition { access_granule, .. } = param.partition();
+
+        match access_granule {
+            Granularity::B32 => {
+                let data_iter = data.iter().take(size as usize);
+                for (idx, data_word) in data_iter.enumerate() {
+                    let addr = byte_addr + (idx * mem::size_of::<u32>()) as u32;
+                    Self::dmi_write(jtag, addr, access_granule, [*data_word, 0x00])?;
+                }
+            }
+            Granularity::B64 => {
+                let data_iter = data.chunks(2).take(size as usize / 2);
+                for (idx, data_words) in data_iter.enumerate() {
+                    let addr = byte_addr + (idx * mem::size_of::<u64>()) as u32;
+                    Self::dmi_write(jtag, addr, access_granule, [data_words[0], data_words[1]])?;
+                }
+            }
+        }
+
+        Ok(())
+    }
+
+    /// Perform a single read over the Direct Memory Interface.
```

Comment:
I believe its DAI = [Direct Access Interface](https://opentitan.org/book/hw/ip/otp_ctrl/doc/theory_of_operation.html#direct-access-interface-control)
```suggestion
    /// Perform a single read over the Direct Access Interface.
```